### PR TITLE
SSCS-2761: smoke/functional tests

### PR DESCRIPTION
**Pull request by paulgain**
* Complete rewrite of the original e2e tests that previously used a
  local DB.
* Added 3 small smoke tests:
    1. appeal received smoke test
    2. health check
    3. surname cookies test
* Added a comprehensive set of functional tests that target the critical
  areas of TYA.
* Wrote a nunjucks helper which translates content on the fly.
* Moved all express paths into a path.js file
* Removed unit/templates/trackYourAppeal.test.js as we now apply the
  same test within our functional tests.

### CODE REVIEW

File: test/e2e/functional/links/links.test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  I.seeCurrentUrlEquals(`${paths.tya.contactus}/${appeal.appealNumber}`);
+
+});
+
+Scenario('Clicking the \'Cookies\' link ', function*(I) {
+
+  I.click(cookiePolicy.footer.cookies);
+  I.see(cookiePolicy.cookies.title);
+  I.seeCurrentUrlEquals(paths.tya.cookiepolicy);
+});
```
Comment (bendiggle) [inline test/e2e/functional/links/links.test.js:48]: I don't think any of these need to be generator functions so you can change them to `(I) => {}`

File: test/e2e/functional/ga/google-analytics.test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  I.seeInSource('<script src="/public/javascripts/google-analytics-universal-tracker.js"></script>');
+
+});
+
+Scenario('Verify UA account code has been loaded', function*(I) {
+
+  I.amOnPage('/public/javascripts/tya-analytics-tracker.js');
+  I.seeInSource('universalId: \'UA-91309785-1\'');
+
+});
```
Comment (bendiggle) [inline test/e2e/functional/ga/google-analytics.test.js:21]: I don't think any of these need to be generator functions so you can change them to `(I) => {}`

File: test/e2e/functional/page/aboutHearing.test.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  // Hearing image caption
+  I.see(env.renderString(hearing.expectations.theHearingRoom.caption, benefitType));
+
+  // DWP
+  I.see(env.renderString(hearing.expectations.dwp.content[0], benefitType));
+
+  // Getting a decision
+  I.see(env.renderString(hearing.expectations.gettingDecision.content[1], benefitType));
+
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/aboutHearing.test.js:30]: Same as above (generator function)

File: test/e2e/functional/page/contactUs.test.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  I.see(contactUs.description);
+  I.see(contactUs.openingHours);
+  I.see(contactUs.englandWales.heading);
+
+  // Scotland content
+  I.see(contactUs.englandWales.phoneNumber);
+  I.see(contactUs.scotland.heading);
+  I.see(contactUs.scotland.phoneNumber);
+
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/contactUs.test.js:28]: Same as above (generator function)

File: test/e2e/functional/page/tya/adjourned.test.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  I.enterSurnameAndSubmitAndSeeTYA(appeal);
+});
+
+Scenario('Verify adjourned appeal details, progress bar status and screen reader text', function*(I) {
+
+  I.seeAppealDetails(appeal);
+  I.seeProgressBarAtDWPRespond();
+  I.seeScreenReaderTextAtDWPRespond();
+
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/tya/adjourned.test.js:15]: Same as above (generator function)

File: test/e2e/functional/page/tya/closed.test.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+Scenario('Verify closed appeal details, no progress bar and content', function*(I) {
+
+  I.seeAppealDetails(appeal);
+  I.dontSeeAProgressBar();
+
+  // Content specific to Closed.
+  I.see(env.renderString(closed.content[0], { benefitType: appeal.benefitType }));
+  I.see(closed.content[1]);
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/tya/closed.test.js:21]: Same as above (generator function)

File: test/e2e/functional/page/tya/dormant.test.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+Scenario('Verify dormant appeal details, progress bar status and screen reader text', function*(I) {
+
+  I.seeAppealDetails(appeal);
+  I.seeProgressBarAtHearing();
+  I.seeScreenReaderTextAtHearing();
+
+});
+
+
+
```
Comment (bendiggle) [inline test/e2e/functional/page/tya/dormant.test.js:18]: Same as above (generator function)

File: test/e2e/functional/page/tya/dwpRespond.test.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  I.enterSurnameAndSubmitAndSeeTYA(appeal);
+});
+
+Scenario('Verify DWP respond appeal details, progress bar status and screen reader text', function*(I) {
+
+  I.seeAppealDetails(appeal);
+  I.seeProgressBarAtDWPRespond();
+  I.seeScreenReaderTextAtDWPRespond();
+
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/tya/dwpRespond.test.js:15]: Same as above (generator function)

File: test/e2e/functional/page/tya/dwpRespondOverdue.test.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  I.enterSurnameAndSubmitAndSeeTYA(appeal);
+});
+
+Scenario('Verify DWP respond overdue appeal details, progress bar status and screen reader text', function*(I) {
+
+  I.seeAppealDetails(appeal);
+  I.seeProgressBarAtAppealReceived();
+  I.seeScreenReaderTextAtAppealReceived();
+
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/tya/dwpRespondOverdue.test.js:15]: Same as above (generator function)

File: test/e2e/functional/page/tya/hearing.test.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+  I.enterSurnameAndSubmitAndSeeTYA(appeal);
+});
+
+Scenario('Verify hearing appeal details, progress bar status and screen reader text', function*(I) {
+
+  I.seeAppealDetails(appeal);
+  I.seeProgressBarAtHearing();
+  I.seeScreenReaderTextAtHearing();
+
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/tya/hearing.test.js:15]: Same as above (generator function)

File: test/e2e/functional/page/tya/hearingBooked.test.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  I.enterSurnameAndSubmitAndSeeTYA(appeal);
+});
+
+Scenario('Verify hearing booked appeal details, progress bar status and screen reader text', function*(I) {
+
+  I.seeAppealDetails(appeal);
+  I.seeProgressBarAtHearingBooked();
+  I.seeScreenReaderTextAtHearingBooked();
+
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/tya/hearingBooked.test.js:15]: Same as above (generator function)

File: test/e2e/functional/page/tya/lapsedRevised.test.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+Before((I) => {
+  I.enterSurnameAndSubmitAndSeeTYA(appeal);
+});
+
+Scenario('Verify lapsed revised appeal details and no progress bar', function*(I) {
+
+  I.seeAppealDetails(appeal);
+  I.dontSeeAProgressBar();
+
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/tya/lapsedRevised.test.js:14]: Same as above (generator function)

File: test/e2e/functional/page/tya/newHearingBooked.test.js [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+  I.enterSurnameAndSubmitAndSeeTYA(appeal);
+});
+
+Scenario('Verify new hearing booked appeal details, progress bar status and screen reader text', function*(I) {
+
+  I.seeAppealDetails(appeal);
+  I.seeProgressBarAtHearingBooked();
+  I.seeScreenReaderTextAtHearingBooked();
+
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/tya/newHearingBooked.test.js:15]: Same as above (generator function)

File: test/e2e/functional/page/tya/pastHearingBooked.test.js [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+  I.enterSurnameAndSubmitAndSeeTYA(appeal);
+});
+
+Scenario('Verify past hearing booked appeal details, progress bar status and screen reader text', function*(I) {
+
+  I.seeAppealDetails(appeal);
+  I.seeProgressBarAtDWPRespond();
+  I.seeScreenReaderTextAtDWPRespond();
+
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/tya/pastHearingBooked.test.js:15]: Same as above (generator function)

File: test/e2e/functional/page/tya/postponed.test.js [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+  I.enterSurnameAndSubmitAndSeeTYA(appeal);
+});
+
+Scenario('Verify postponed appeal details, progress bar status and screen reader text', function*(I) {
+
+  I.seeAppealDetails(appeal);
+  I.seeProgressBarAtDWPRespond();
+  I.seeScreenReaderTextAtDWPRespond();
+
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/tya/postponed.test.js:15]: Same as above (generator function)

File: test/e2e/functional/page/tya/withdrawn.test.js [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+Scenario('Verify appeal details, no progress bar and content', function*(I) {
+
+  I.seeAppealDetails(appeal);
+  I.dontSeeAProgressBar();
+
+  // Content specific to Withdrawn.
+  I.see(content.common.latestUpdate);
+  I.see(content.status.withdrawn.content[0]);
+  I.see(content.status.withdrawn.content[1]);
+});
```
Comment (bendiggle) [inline test/e2e/functional/page/tya/withdrawn.test.js:19]: Same as above (generator function)

**Review by bendiggle — CHANGES_REQUESTED** [review]

File: test/e2e/functional/links/links.test.js (hunk 1 above)
Comment (paulgain) [reply test/e2e/functional/links/links.test.js:48]: Good spot.

File: test/e2e/functional/page/aboutHearing.test.js (hunk 3 above)
Comment (paulgain) [reply test/e2e/functional/page/aboutHearing.test.js:30]: yep

File: test/e2e/functional/page/contactUs.test.js (hunk 4 above)
Comment (paulgain) [reply test/e2e/functional/page/contactUs.test.js:28]: yep

**Review by bendiggle — APPROVED** [review]

**Pull request merged**